### DISCUSSION TITLE
font-size as a variable

### DISCUSSION THREAD

**Discussion by AlexStuffs** [jspsych/jsPsych#1199] [Q&A] [answered]
Hi all, 

I'm looking for a way to make the font-size of a stimulus callable from a variable. 
(In fact, I'm trying to compute a version of a SART)

I'm a beginner in both jsPsych and scripting, I see several ways to achieve this but I may wrong. 
I think I can use jsPsych.randomization.factorial with something like this:
`var factors = {
      stimulus: [
        {stimulus: '1', correct_response: ['space']},
        {stimulus: '2', correct_response: ['space']}],
      font_size: [48,120]
    }
    var full_design = jsPsych.randomization.factorial(factors, 1);`

However, it could be quite heavy (I'll have 9 numbers and 4 to 5 font-sizes) and I don't know how to call both the number and its font-size. 

I think it may be possible to use timelineVariables in order to randomize the font-size in the same way that it's shown in the demo reaction time tuto for trial_duration.
However, I can't figure out how to insert both the number and the font-size in the `stimulus `parameter...

I guess that someone will say that there sooooooo much easier way to make that. 
And I'll appreciate if it's understandable for me ^^'

Thanks!

Alex

**Comment by becky-gilbert** [comment, marked as answer]
Hi @AlexStuffs I'm not 100% I understand what you're trying to do, but in response to this question:
> However, I can't figure out how to insert both the number and the font-size in the stimulus parameter...

You can use a [function for your stimulus](https://www.jspsych.org/overview/trial/#dynamic-parameters), and inside of that function, get both the number and font size using `jsPsych.timelineVariable()` with the second `true` parameter, and return a value for the stimulus that combines these things:
```javascript
// set up your timeline_variables array - each object contains the information for 1 trial
var stimuli_info = [ 
  {stimulus: '1', correct_response: ['space'], font_size: 48}, 
  {stimulus: '2', correct_response: ['space'], font_size: 48},
  {stimulus: '1', correct_response: ['space'], font_size: 120}, 
  {stimulus: '2', correct_response: ['space'], font_size: 120},
];

var trial = {
  type: 'html-keyboard-response',
  choices: jsPsych.ALL_KEYS,
  stimulus: function() { 
    // this is how you can combine different variables from your timeline_variables array
    return '<p style="font-size: '+jsPsych.timelineVariable('font-size', true)+'px;">'+jsPsych.timelineVariable('stimulus', true)+'</p>';
  },
  data: {correct_response: jsPsych.timelineVariable('correct_response')
}

var trial_procedure = {
  timeline: [trial],
  timeline_variables: stimuli_info
};
```
This is also explained in the second example in the [timeline variables docs](https://www.jspsych.org/overview/timeline/#timeline-variables). Does that help at all?

**Reply by AlexStuffs** [reply]
Hi @becky-gilbert ,
Thank you for your reply!
I remember trying something similar but I can't remember why it didn't work...
I turned my code in another way but your code will greatly simplify my work!

Thanks!
